src/walletSdk/Asset/index.ts

```diff
@@ -0,0 +1,58 @@
+import StellarSdk from "stellar-sdk";
+
+const STELLAR_SCHEME = "stellar";
+
+const FIAT_SCHEME = "iso4217";
+
+class AssetId {
+  id: string;
+  scheme: string;
+  get sep38() {
+    return `${this.scheme}:${this.id}`;
+  }
+}
+
+class StellarAssetId extends AssetId {
+  id: string;
+  code: string;
+  issuer: string;
+  scheme = STELLAR_SCHEME;
+
+  toAsset() {
+    if (this.id === "native") {
+      return new StellarSdk.Asset("XLM");
+    }
+    return new StellarSdk.Asset(this.code, this.issuer);
+  }
+}
+
+export class IssuedAssetId extends StellarAssetId {
+  constructor(code: string, issuer: string) {
+    super();
+    this.code = code;
+    this.issuer = issuer;
+    this.id = `${this.code}:${this.issuer}`;
+  }
+
+  toString() {
+    return this.sep38;
+  }
+}
+
+export class NativeAssetId extends StellarAssetId {
+  id = "native";
+  code = "XLM";
+}
+
+export class FiatAssetId extends AssetId {
+  id: string;
+  scheme = FIAT_SCHEME;
+  constructor(id: string) {
```

Comment:
What do you think about renaming this constructor input to `code` instead of `id`?
```
  constructor(code: string) {
    super();
    this.id = code;
  }
```

I personally think it'd be more intuitive since for fiat we use code as the id, and asset id always makes me think it's something more complex with a `:` in it

Reply:
ya I agree with that, changed https://github.com/stellar/typescript-wallet-sdk/pull/34/commits/434ec91051ce906e07d0e41802609cab8a0b2740